tests/system_tests_topology.py

```diff
@@ -592,5 +594,182 @@ def run(self):
+        INT_B.inter_router_port = i_r_port
+
+        INT_C = self._create_router('INT.C',
+                                    ra_interval,
+                                    ra_stale,
+                                    ra_flux,
+                                    extra=[('connector',
+                                            {'role': 'inter-router',
+                                             'name': 'connectorToB',
+                                             'port': i_r_port})])
+        #
+        # wait until router network is formed
+        #
+        INT_B.wait_router_connected('INT.A')
+        INT_B.wait_router_connected('INT.C')
+
+        #
+        # create mobile addresses on INT_A
+        #
+        consumers = [
+            AsyncTestReceiver(INT_A.listener,
+                              source='closest/on_A'),
+            AsyncTestReceiver(INT_A.listener,
+                              source='closest/on_A')]
+        #
+        # wait for addresses to show up on INT.C
+        #
+        INT_C.wait_address('closest/on_A')
+
+        return (INT_A, INT_B, INT_C, consumers)
+
+    def test_01_reboot_INT_A(self):
+        """
+        When a router comes online after a reboot its route table sequence will
+        be different from the last update it sent.  This should cause the local
+        router to flush all mobile addresses it learned from the remote router
+        before it rebooted.
+
+        Reboot INT.A and expect its mobile addresses are flushed on INT_C
+        """
+
+        # bump the remoteLsMaxAgeSeconds to longer than the test timeout so the
+        # test will timeout if the addresses are not removed before the link is
+        # considered stale
+        stale_timeout = int(TIMEOUT * 2)
+        INT_A, INT_B, INT_C, consumers = self._deploy_routers(ra_stale=stale_timeout)
+
+        # at this point all routers are running and the mobile addresses have
+        # propagated to INT_C.  Now reboot INT_A
+        INT_A.teardown()
+        time.sleep(1.0)
+        INT_A = self._create_router('INT.A',
+                                    ra_stale=stale_timeout,
+                                    extra=[('connector',
+                                            {'role': 'inter-router',
+                                             'name': 'connectorToB',
+                                             'port':
+                                             INT_B.inter_router_port})])
+        INT_A.wait_router_connected('INT.B')
+
+        # wait for INT_A mobile addresses to be removed from INT_C
+        mgmt = INT_C.management
+        a_type = 'org.apache.qpid.dispatch.router.address'
+        rsp = mgmt.query(a_type).get_dicts()
+        while any(map(lambda a: a['name'].find('closest/on_A') != -1, rsp)):
+            time.sleep(0.25)
+            rsp = mgmt.query(a_type).get_dicts()
+
+        # cleanup
+        for c in consumers:
```

Comment:
Can we please move this loop to close the consumer right after the INT_A.teardown() call just to be clear that the consumers are gone forever and will not try to reconnect after the router is bounced.

Reply:
done